algosdk/future/atomic_transaction_composer.py

```diff
@@ -0,0 +1,533 @@
+            raise error.AtomicTransactionComposerError(
+                "AtomicTransactionComposer must be in BUILDING state for a transaction to be added"
+            )
+        if (
+            len(self.txn_list) + method_call.get_txn_calls()
+            > self.MAX_GROUP_SIZE
+        ):
+            raise error.AtomicTransactionComposerError(
+                "AtomicTransactionComposer cannot exceed MAX_GROUP_SIZE transactions"
+            )
+        if method_call.args and len(method_call.args) != len(method_args):
+            raise error.AtomicTransactionComposerError(
+                "number of method arguments do not match the method signature"
+            )
+        if not isinstance(method_call, abi.method.Method):
+            raise error.AtomicTransactionComposerError(
+                "invalid Method object was passed into AtomicTransactionComposer"
+            )
+
+        app_args = []
+        foreign_accounts = []
+        foreign_assets = []
+        foreign_apps = []
+        # For more than 14 args, including the selector, compact them into a tuple
+        additional_args = []
+        additional_types = []
+        txn_list = []
+        # First app arg must be the selector of the method
+        app_args.append(method_call.get_selector())
+        # Iterate through the method arguments and either pack a transaction
+        # or encode a ABI value.
+        for i, arg in enumerate(method_call.args):
+            if arg.type in abi.method.TRANSACTION_ARGS:
+                if not isinstance(method_args[i], TransactionWithSigner):
+                    raise error.AtomicTransactionComposerError(
+                        "expected TransactionWithSigner as method argument, but received: {}".format(
+                            method_args[i]
+                        )
+                    )
+                txn_list.append(method_args[i])
+            else:
+                if arg.type in abi.method.FOREIGN_ARRAY_ARGS:
+                    current_type = abi.UintType(8)
+                    if arg.type == "account":
+                        current_arg = len(foreign_accounts)
+                        foreign_accounts.append(method_args[i])
+                    elif arg.type == "asset":
+                        current_arg = len(foreign_assets)
+                        foreign_assets.append(method_args[i])
+                    else:
+                        current_arg = len(foreign_apps)
+                        foreign_apps.append(method_args[i])
+                else:
+                    current_type = arg.type
+                    current_arg = method_args[i]
+                if len(app_args) > 14:
+                    # Pack the remaining values as a tuple
+                    additional_types.append(current_type)
+                    additional_args.append(current_arg)
+                else:
+                    encoded_arg = arg.type.encode(current_arg)
```

Comment:
Getting `TypeError: encode() argument 'encoding' must be str, not int`  here for an 'account' type. 

Looks like this attribute is a `str` not one of the ABI types  https://github.com/algorand/py-algorand-sdk/blob/algochoi/foreign-types/algosdk/abi/method.py#L149

Reply:
My bad, that should be: 
`encoded_arg = current_type.encode(current_arg)` 

I will push this change in the short term and add some more tests after the other branch is merged.